### PR TITLE
Create tesla_mcu_defconfig

### PR DESCRIPTION
**Pull request by jnuyens**
Added the current kernel config for easier hacking and reference.

### CODE REVIEW

**Comment by deftdawg** [comment]
@jnuyens  your kernel config is from a rooted car, they do actually provide kernel configs for the cid and ic in /arch/arm/configs/ts001_defconfig and /arch/arm/configs/ts001_ic_defconfig... 

There are a number of unspecified options, so maybe you're supposed to do:
```
make ts001_defconfig ARCH=arm
yes "" | make oldconfig
make -j4
```
You need tesla.cpio, which I assume is the product of a successful buildroot...  have you had any luck building either?

**EDIT:** Hmm...  looks like I need ARCH=arm every time I invoke make...

**Comment by deftdawg** [comment]
From taking a look at the [build doc](https://github.com/teslamotors/linux/blob/tegra-4.4/codethink/documents/build.txt), I'm guessing that they compile natively on an arm system, since they don't mention arch or any cross compile stuff.

**Comment by deftdawg** [comment]
I think this should work, once a proper build root "tesla.cpio" can be produced:
```
apt install gcc-arm-linux-gnueabi libncurses5-dev
alias make='ARCH=arm make'
make ts001_defconfig
make -j4 CROSS_COMPILE=arm-linux-gnueabi- Image zImage tegra30-tesla-cid.dtb
```

**Comment by jnuyens** [comment]
It's a correct answer, but I prefer not to advise to make an alias like that. Such an alias only remains active in the current session, and you don't want to have a persistent make alias like that because it will mean that on your host system things could go very wrong, also it hides what is actually being done. Just specify ARCH=arm with any make command you issue: 

>apt install gcc-arm-linux-gnueabi libncurses5-dev 
>make ARCH=arm ts001_defconfig 
>make ARCH=arm -j4 CROSS_COMPILE=arm-linux-gnueabi- Image zImage tegra30-tesla-cid.dtb

**Comment by deftdawg** [comment]
I would guess there is no further need for this patch as the kernel config already exists in the source tree...  through the documentation could use an update to indicate that it's already there.

**Review by crypt096 — APPROVED** [review]

File: arch/arm/configs/tesla_defconfig [hunk 1]
```diff
@@ -0,0 +1,3510 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# Linux/arm 4.4.35 Kernel Configuration
+#
+CONFIG_ARM=y
+CONFIG_ARM_HAS_SG_CHAIN=y
+CONFIG_MIGHT_HAVE_PCI=y
+CONFIG_SYS_SUPPORTS_APM_EMULATION=y
+CONFIG_HAVE_PROC_CPU=y
+CONFIG_STACKTRACE_SUPPORT=y
```
Comment (Pines408) [inline arch/arm/configs/tesla_defconfig]: My assets

Comment (Pines408) [reply arch/arm/configs/tesla_defconfig]: how can i get my assets

**Pull request closed without merging**